### ISSUE TITLE
Error after 6.01, definitions.json not found

### ISSUE THREAD

**Issue by YOxidized** [Soreepeong/XivMitmLatencyMitigator#30]
Probably aware of this issue, but just in case, when I start this, I receive the following output that may be related to a change in the XIVAlex source:

![image](https://user-images.githubusercontent.com/79140667/146908185-93ce4ae8-8ea6-4ea4-b11d-fa82c3a6fe9b.png)

**Comment by Soreepeong** [comment]
Try giving `-u` argument when running the script for now.

**Comment by YOxidized** [comment]
No problem!  Just wanted to make sure this was accounted for.

**Comment by BloodletterXIV** [comment]
Hello, a friend told me xivalex was up so I decided to try and run this again too, however I'm finding the same issue posted above.

I don't understand much about coding so I don't get what it means to give the -u argument when running the script, is this done in the iptables line?

**Comment by Soreepeong** [comment]
@BloodletterXIV Try running `rm definitions.json`, and ignore the result. After that, run the last line `curl ... | python` again. It should work now, but I couldn't test it yet.

**Comment by YOxidized** [comment]
I do not believe it is working for me, I get this output, but the lag in game persists nor am I getting a log of my actions:

![image](https://user-images.githubusercontent.com/79140667/146956220-f60006c3-f5e3-42ae-97f5-b0829e534c4a.png)

**Comment by YOxidized** [comment]
It is connecting, but the actual latency mitigation feels like it is not going through.  I am no longer seeing a "DelayAdjusted" value on my log, and can still barely double weave without clipping, suggesting it is not picking up the latency correction.  Here is what my log is reporting when I am queueing and performing actions:

![image](https://user-images.githubusercontent.com/79140667/146960198-b83fe0eb-a2dd-4ca0-b8b9-b82427b681be.png)

**Comment by deepwebcactus** [comment]
im having the exact same issue as YOxidized, its connecting but not mitigating

**Comment by BloodletterXIV** [comment]
@Soreepeong I did exactly as instructed and I'm having the same problem as the users above, I can get into the game but double weaving is impossible. I noticed there was an error and a "shutdown" the moment I got past the queue and loaded into the game, which I've never seen before at that part.

**Comment by Soreepeong** [comment]
@YOxidized You might want to try fiddling with `-e [fixed extra delay in seconds]` to set different extra delay, or `-m` to turn ping measuring on again. Measured ping tend to be largely inaccurate, so I turned it off by default.

@BloodletterXIV That's probably an another problem. Screencap of the error messages would help.

**Comment by YOxidized** [comment]
These changes seem to have a very adverse effect on the program's effectiveness, and the new log reporting makes it very difficult to really measure the discrepancy.  I can try these fixes, but I may just fork the version before yesterday's changes instead.

**Comment by BloodletterXIV** [comment]
@Soreepeong Alright, here's what's happening exactly:

After the iptables line, I start the script and it gives me this (the opcode downloading stuff is totally new, this never happened in 3 months and when EW came out I just did `rm definitions.json` and everything worked perfectly).

![01](https://user-images.githubusercontent.com/91862122/146971158-a6ff2dea-614d-4850-852b-a804cc0d72c7.png)

Once I did that I stopped the program with Ctrl+C and it gave me this "cleaning up" log which is also something new. The `rm definitions.json` seemed to work fine after.

![02](https://user-images.githubusercontent.com/91862122/146971410-56cee0b7-bcc8-4450-ab0b-a84ff855ee9c.png)

Finally, I ran the curl line again and got into the game. `The D->U Read Finish` below is when I loaded in and the very last part was just an example of what it shows when I perform an OGCD action.

![03](https://user-images.githubusercontent.com/91862122/146971783-8b8e48da-84e8-4199-8316-29522d75367c.png)

**Comment by YOxidized** [comment]
@BloodletterXIV Your issues seem to mirror my own, so I do not think it is anything to do with your VM; I appreciate you thoroughly explaining them in this manner, and your screenshots are exactly what I am seeing in my VM.

**Comment by deepwebcactus** [comment]
my log is the same as BloodletterXIV

**Comment by deepwebcactus** [comment]
![curl](https://user-images.githubusercontent.com/96489100/146978268-43aeba59-6049-40b1-9219-1fe28362dd6f.png)
 thats what appears when i input the curl line (which never appeared like that)
 
 
![login actions](https://user-images.githubusercontent.com/96489100/146978367-1ae873b4-68c0-4030-a1ed-d50a9de4e841.png)
 and thats me logging in and using gcds and ogcds in a  dummy

**Comment by Soreepeong** [comment]
Okay, I think I figured out why it didn't do anything. I was misunderstanding that slicing a bytearray will return a partial view, but it actually returned a copy. It should correctly mitigate stuff now.

**Comment by deepwebcactus** [comment]
I can confirm, that at least for me, it is working properly now, thanks and keep the amazing work!

**Comment by YOxidized** [comment]
It does work, but it still lacks the log reporting that lets the user know exactly what is going on in the background, and though I have nothing to substantiate this claim, feels slightly less effective.  Thank you for your hard work as always, but may I ask what benefits this new version has over the previous?

**Comment by Soreepeong** [comment]
Previous version by default tended to overcompensate in a lot of cases, due to inaccurate latency measurement, so it's off by default. You can turn it on by giving it a flag.

Instead of doing `curl http://.... | python`, try:

```
curl http://... -o mitigate.py
python mitigate.py -m
```

**Comment by YOxidized** [comment]
Ah, if it overcompensated, then that makes sense to change.  I will experiment with -m toggled on, but this version still does a fine job of achieving the desired results.  I may experiment with toggling the -m option, and thank you so much once again for your hard work, dedication, and for developing this for PS4 players so that we may share a raid comfort than many PC players have access to.

**Comment by Soreepeong** [comment]
If you ever noticed skills being refunded not because of distance nor enemy becoming untargetable, that is a symptom of overcompensation. If it never happened to you, -m is probably fine to use for your case.

**Comment by YOxidized** [comment]
This is really good to know; I have heard of people on XIVAlex experiencing this in the past, but it has not once happened to me, so I will utilize the -m flag for now and disable it if I see any of this behavior as a result of its usage.
